internal/grpc/services/gateway/gateway.go

```diff
@@ -116,10 +117,11 @@ func (c *config) init() {
 }
 
 type svc struct {
-	c              *config
-	dataGatewayURL url.URL
-	tokenmgr       token.Manager
-	etagCache      *ttlcache.Cache `mapstructure:"etag_cache"`
+	c                 *config
+	dataGatewayURL    url.URL
+	tokenmgr          token.Manager
+	etagCache         *ttlcache.Cache `mapstructure:"etag_cache"`
+	hiddenRootFolders map[string]map[string]struct{}
```

Comment:
Let's just store the map as passed by the user. At max, the list will contain 5 elements, searching over those won't lead to any performance drawback